src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommandRunner.cs

```diff
@@ -0,0 +1,450 @@
+            }
+            else if (!outdated && printingTransitive)
+            {
+                tableToPrint = packages.ToStringTable(
+                        headers,
+                        p => "",
+                        p => p.name,
+                        p => "",
+                        p => p.resolvedVersion.ToString());
+            }
+            else
+            {
+                tableToPrint = packages.ToStringTable(
+                       headers,
+                       p => "",
+                       p => p.name,
+                       p => {
+                           if (p.autoReference)
+                           {
+                               autoReferenceFound = true;
+                               return "(A)";
+                           }
+                           return "";
+                       },
+                       p => p.printableRequestedVersion,
+                       p => p.resolvedVersion.ToString());
+            }
+
+            //Handle printing with colors
+            foreach (var line in tableToPrint)
+            {
+                if (line.Item2 != ConsoleColor.White)
+                {
+                    Console.ForegroundColor = line.Item2;
+                }
+
+                Console.WriteLine(line.Item1);
+                Console.ResetColor();
+            }
+
+            Console.WriteLine();
+            autoRefWithinPackagesList = autoReferenceFound;
+        }
+
+        /// <summary>
+        /// Prepares the headers for the tables that will be printed
+        /// </summary>
+        /// <param name="printingTransitive">Whether the table is for transitive or not</param>
+        /// <param name="outdated">Whether we have an outdated/latest column or not</param>
+        /// <returns></returns>
+        private string[] BuildTableHeaders(bool printingTransitive, bool outdated)
+        {
+            var result = new List<string> { LeftPadding };
+            if (printingTransitive)
+            {
+                result.Add("Transitive Package");
+                result.Add("");
+                result.Add("Resolved");
+            }
+            else
+            {
+                result.Add("Top-level Package");
+                result.Add("");
+                result.Add("Requested");
+                result.Add("Resolved");
+            }
+
+            if (outdated)
+            {
+                result.Add("Latest");
+            }
+
+            return result.ToArray();
+        }
+    }
+
+    internal struct FrameworkPackages
+    {
+        public string framework;
+        public IEnumerable<PRPackage> topLevelPackages;
+        public IEnumerable<PRPackage> transitivePacakges;
+    }
+
+    /// <summary>
+    /// A struct to simplify holding all of the information
+    /// about a package reference when using list
+    /// </summary>
+    internal struct PRPackage
```

Comment:
PRPackage -> change this to InstalledPackageReference or something like that. Avoid using short forms in class names.